apps/ios/Shared/Views/UserSettings/SMPServerView.swift

```diff
@@ -24,47 +56,61 @@ struct SMPServerView: View {
         return VStack {
             List {
                 Section("Preset server address") {
-                    Text(server.server)
+                    Text(serverToEdit.server)
+                        .textSelection(.enabled)
                 }
-                useServerSection()
+                useServerSection(true)
             }
         }
     }
 
     private func customServer() -> some View {
         VStack {
+            let valid = parseServerAddress(serverToEdit.server)?.valid == true
             List {
-                Section("Your server address") {
-                    TextEditor(text: $server.server)
+                Section {
+                    TextEditor(text: $serverToEdit.server)
                         .multilineTextAlignment(.leading)
                         .autocorrectionDisabled(true)
                         .autocapitalization(.none)
+                        .allowsTightening(true)
                         .lineLimit(10)
-                        .frame(height: 108)
+                        .frame(height: 144)
                         .padding(-6)
+                } header: {
+                    HStack {
+                        Text("Your server address")
+                        if !valid {
+                            Spacer()
+                            Image(systemName: "exclamationmark.circle").foregroundColor(.red)
+                        }
+                    }
                 }
-                useServerSection()
+                useServerSection(valid)
                 Section("Add to another device") {
-                    QRCode(uri: server.server)
+                    MutableQRCode(uri: $serverToEdit.server)
                         .listRowInsets(EdgeInsets(top: 12, leading: 12, bottom: 12, trailing: 12))
                 }
             }
         }
     }
 
-    private func useServerSection() -> some View {
+    private func useServerSection(_ valid: Bool) -> some View {
         Section("Use server") {
             HStack {
                 Button("Test server") {
-                    Task { await testServerConnection(server: $server) }
+                    Task {
```

Comment:
in this view progress indicator is not displayed when testing a previously tested server

Reply:
actually on a new server as well